### PR TITLE
PROD-178 - add-delete-deck-functionality-to-admin-panel

### PR DESCRIPTION
**Pull request by danijel-radomiljac**

### CODE REVIEW

File: app/actions/deck/deck.ts [hunk 1]
```diff
@@ -15,6 +15,154 @@ import { formatErrorsToString } from "../../utils/zod";
 import { handleUpsertingQuestionOptionsConcurrently } from "../question/question";
 import { deckInputFactory } from "./factories";
 
+export async function deleteQuestions(questionIds: number[]) {
+  const isAdmin = await getIsUserAdmin();
+
+  if (!isAdmin) {
+    redirect("/application");
```
Comment (marvinmarnold) [inline app/actions/deck/deck.ts:22]: Is redirecting here sufficient? Shouldn't we `return` to ensure execution doesn't continue?

File: app/api/deck/route.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { getTotalNumberOfAnswersInDeck } from "@/app/actions/deck/deck";
+
+export async function GET(request: Request) {
+  const url = new URL(request.url);
+  const deckId = Number(url.searchParams.get("deckId"));
+
+  if (isNaN(deckId)) {
```
Comment (marvinmarnold) [inline app/api/deck/route.ts:7]: Can we also protect with `getIsUserAdmin`?

File: app/actions/deck/deck.ts (hunk 1 above)
Comment (danijel-radomiljac) [reply app/actions/deck/deck.ts:22]: it also causes return

File: app/api/deck/route.ts (hunk 2 above)
Comment (marvinmarnold) [reply app/api/deck/route.ts:7]: Can you please add anyways? I prefer to expose what's minimally necessary.

File: app/actions/deck/deck.ts (hunk 1 above)
Comment (marvinmarnold) [reply app/actions/deck/deck.ts:22]: Yea, you are right `Invoking the redirect() function throws a NEXT_REDIRECT error and terminates rendering of the route segment in which it was thrown.`

File: app/api/deck/route.ts (hunk 2 above)
Comment (danijel-radomiljac) [reply app/api/deck/route.ts:19]: added ✅

**Review by marvinmarnold — APPROVED** [review]

**Pull request merged**